schemas/theme/theme_block.json

```diff
@@ -106,45 +106,45 @@
     }
   },
   "definitions": {
-    "blocks_array": {
+    "blocksArray": {
       "type": "array",
       "description": "A list of child blocks that you might want to include.",
       "markdownDescription": "A list of child blocks that you might want to include.\n\n---\n\n[Shopify reference](https://shopify.dev/docs/themes/architecture/blocks/theme-blocks/schema#presets)",
-      "required": ["type"],
       "items": {
         "type": "object",
         "properties": {
-          "$ref": "#/definitions/base_block",
-          "id": {
-            "type": "string"
-          }
-        },
-        "allOf": [
-          {
-            "if": {
-              "properties": {
-                "static": {
-                  "const": true
-                }
-              },
-              "required": ["static"]
-            },
-            "then": {
-              "required": ["id"]
+          "allOf": [{ "$ref": "#/definitions/commonBlockAttributes" }],
+          "properties": {
+            "id": {
+              "type": "string",
+              "description": "A unique identifier for the block."
             }
-          }
-        ]
+          },
+          "additionalProperties": false
+        },
+        "if": {
+          "properties": {
+            "static": { "const": true }
+          },
+          "required": ["static"]
+        },
+        "then": {
+          "required": ["id"]
+        }
       }
     },
-    "blocks_hash": {
+    "blocksHash": {
       "type": "object",
       "description": "A list of child blocks that you might want to include.",
       "markdownDescription": "A list of child blocks that you might want to include.\n\n---\n\n[Shopify reference](https://shopify.dev/docs/themes/architecture/blocks/theme-blocks/schema#presets)",
-      "additionalProperties": {
-        "$ref": "#/definitions/base_block"
-      }
+      "patternProperties": {
+        "^[a-zA-Z0-9_-]+$": {
```

Comment:
I looked in the docs, asked AI, and this is the answer I found for an object that can contain an unknown number of objects.

Reply:
I don't understand why we'd want the regex instead of additionalProperties. Those should be equivalent? 🤔 

Only thing pattern properties adds is a validation of the property name. It's not bad to have, but I don't think it's required to have unknown key/value pairs.

Reply:
Fair, I had a feeling it might be overkill.